projects/vivaleva/index.js

```diff
@@ -62,19 +26,26 @@ module.exports = {
         syncswapWorkerBalances,
         syncswapReserves,
         syncswapLpTotalSupplies,
-        token0s,
-        token1s,
       ] = await Promise.all([
-        api.multiCall({ abi: "uint256:totalStakedLpBalance", calls: syncswapWorkers.map((v) => v.address), }),
-        api.multiCall({ abi: "function getReserves() view returns (uint256, uint256)", calls: syncswapWorkers.map((v) => v.lpToken), }),
-        api.multiCall({ abi: "uint256:totalSupply", calls: syncswapWorkers.map((v) => v.lpToken), }),
-        api.multiCall({ abi: "address:token0", calls: syncswapWorkers.map((v) => v.lpToken), }),
-        api.multiCall({ abi: "address:token1", calls: syncswapWorkers.map((v) => v.lpToken), }),
+        api.multiCall({
+          abi: "uint256:totalStakedLpBalance",
+          calls: syncswapWorkers.map((v) => v.address),
+        }),
+        api.multiCall({
+          abi: "function getReserves() view returns (uint256, uint256)",
+          calls: syncswapWorkers.map((v) => v.lpTokenAddress),
+        }),
+        api.multiCall({
+          abi: "uint256:totalSupply",
+          calls: syncswapWorkers.map((v) => v.lpTokenAddress),
+        }),
       ]);
 
       syncswapWorkers.forEach((w, i) => {
-        const token0 = token0s[i]
-        const token1 = token1s[i]
+        const [token0, token1] =
+          w.baseTokenAddress.toLowerCase() < w.farmingTokenAddress.toLowerCase()
```

Comment:
always confused, how does this logic work? isnt is safer to fetch token info from lp?